core/src/main/kotlin/fr/sncf/osrd/stdcm/infra_exploration/InfraExplorer.kt

```diff
@@ -76,7 +76,12 @@ interface EdgeIdentifier {
 }
 
 /** Init all InfraExplorers starting at the given location. */
-fun initInfraExplorer(rawInfra: RawInfra, blockInfra: BlockInfra, location: PathfindingEdgeLocationId<Block>): Collection<InfraExplorer> {
+fun initInfraExplorer(
+    rawInfra: RawInfra,
+    blockInfra: BlockInfra,
+    location: PathfindingEdgeLocationId<Block>,
+    endBlocks: Collection<BlockId> = setOf()
+): Collection<InfraExplorer> {
```

Comment:
Technically, if the path to explore is only on one block (yes this should not happen), you would have to check that the endBlocks do not contain the current block in the init as well, modifying line 96 as follows:
`containsEnd = endBlocks.contains(block),`